src/EndpointManager.sol

```diff
@@ -241,8 +319,29 @@ abstract contract EndpointManager is
             revert ZeroAmount();
         }
 
-        // check outbound rate limits up front
-        _consumeOutboundAmount(amount);
+        if (!shouldSkipRateLimit) {
+            bool didConsumeAmount = _consumeOutboundAmount(amount);
+            if (shouldQueue && !didConsumeAmount) {
+                // queue up and return
+                uint64 queueSequence = useOutboundQueueSequence();
+
+                _getOutboundQueueStorage()[queueSequence] = OutboundQueuedTransfer({
+                    amount: amount,
+                    recipientChain: recipientChain,
+                    recipient: recipient,
+                    txTimestamp: block.timestamp,
+                    isSet: true
+                });
+
+                // refund the price quote back to sender
+                payable(msg.sender).transfer(totalPriceQuote);
```

Comment:
There's an existing refund if `msg.value` exceeds the quote. I'm thinking that should be moved to later in this method because otherwise we're making 2 refunds which costs extra gas